misc/server/Dockerfile

```diff
@@ -1,8 +1,12 @@
-FROM rust:1.66-bullseye as builder
+FROM rust:1.67-bullseye as builder
```

Comment:
I had to bump this in order to compile `libp2p-lookup`

Reply:
While we are at it, why not move to `1.72`, i.e. the most recent version?

Reply:
👍🏻